WireAPI/Sources/WireAPI/APIs/SelfUserAPI/SelfUserAPIV0.swift

```diff
@@ -0,0 +1,135 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class SelfUserAPIV0: SelfUserAPI, VersionedAPI {
+
+    let httpClient: HTTPClient
+
+    init(httpClient: HTTPClient) {
+        self.httpClient = httpClient
+    }
+
+    var apiVersion: APIVersion {
+        .v0
+    }
+
+    func getSelfUser() async throws -> SelfUser {
+        let request = HTTPRequest(
+            path: "\(pathPrefix)/self",
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: 200, type: SelfUserV0.self)
+            .parse(response)
+    }
+
+    func pushSupportedProtocols(_: Set<SupportedProtocol>) async throws {
+        throw SelfUserAPIError.unsupportedEndpointForAPIVersion
+    }
+}
+
+struct SelfUserV0: Decodable, ToAPIModelConvertible {
+
+    let accentID: Int
+    let assets: [UserAsset]?
+    let deleted: Bool?
+    let email: String?
+    let expiresAt: UTCTimeMillis?
```

Comment:
Swagger says this is `UTCTime` not `UTCTimeMillis`.

Reply:
looks like we need UTCTimeMillis as we end up with a decoding error using UTCTime as it expects to decode a date with fractional seconds. the format on Swagger is as follows: 2021-05-12T10:52:02.671Z